zrml/swaps/src/tests.rs

```diff
@@ -46,6 +46,46 @@ const _101: u128 = 101 * BASE;
 const _105: u128 = 105 * BASE;
 const _10000: u128 = 10000 * BASE;
 
+#[test]
+fn destroy_pool_fails_if_pool_does_not_exist() {
+    ExtBuilder::default().build().execute_with(|| {
+        create_initial_pool(ScoringRule::CPMM, true);
+        assert_noop!(Swaps::destroy_pool(42), crate::Error::<Runtime>::PoolDoesNotExist);
+    });
+}
+
+#[test]
+fn destroy_pool_correctly_cleans_up_pool() {
+    ExtBuilder::default().build().execute_with(|| {
+        create_initial_pool_with_funds_for_alice(ScoringRule::CPMM, true);
+        let pool_id = 0;
+        let alice_balance_before = [
+            Currencies::free_balance(ASSET_A, &ALICE),
+            Currencies::free_balance(ASSET_B, &ALICE),
+            Currencies::free_balance(ASSET_C, &ALICE),
+            Currencies::free_balance(ASSET_D, &ALICE),
+        ];
+        assert_ok!(Swaps::destroy_pool(pool_id));
+        assert!(matches!(
+            Swaps::pool_by_id(pool_id),
+            Err(crate::Error::<Runtime>::PoolDoesNotExist)
+        ));
```

Comment:
We should use [`assert_err!`](https://paritytech.github.io/substrate/master/frame_support/macro.assert_err.html) for that.

Reply:
I'm not sure what's going on, but I get 

```
note: an implementation of `PartialEq` might be missing for `pallet::Error<mock::Runtime>`
```

I'm surprised this is an issue. I could add `#[derive(PartialEq)]` to the `Error` enum, but I have a feeling that something else is off.

Reply:
Oh, ok, `DispatchError` implements `PartialEq` by comparing module and error index; but `pool_by_id` returns `Error<T>`.